### PR TITLE
Set/fix typehints in visualize/profiles.py / visualize/reference_poin…

### PR DESCRIPTION
**Pull request by dweindl**
…ts.py

And remove redundant / contradictory types from docstrings

### CODE REVIEW

**Review by Doresic — APPROVED** [review]
Nice changes, thank you.
For my information, descriptions like `profile_indices: list of integer values` are not nice, so that's why you're removing and adding them to the function input like `Sequence[int]`?

File: pypesto/visualize/profiles.py [hunk 1]
```diff
@@ -15,50 +15,50 @@ def profiles(
     results: Union[Result, Sequence[Result]],
     ax=None,
     profile_indices: Sequence[int] = None,
-    size: Sequence[float] = (18.5, 6.5),
+    size: tuple[float, float] = (18.5, 6.5),
     reference: Union[ReferencePoint, Sequence[ReferencePoint]] = None,
     colors=None,
```
Comment (Doresic) [inline pypesto/visualize/profiles.py:20]: Should this be `colors: Sequence[tuple]` or something similar?

**Review by Doresic — APPROVED** [review]

**Comment by dweindl** [comment]
> For my information, descriptions like `profile_indices: list of integer values` are not nice, so that's why you're removing and adding them to the function input like `Sequence[int]`?

They are not bad per se. However, proper type annotations can be used for static code analysis, those other comments cannot. And redundancy is bad. Also, as you can see in some of the changes, those comments don't reflect the actually expected types.

File: pypesto/visualize/profiles.py (hunk 1 above)
Comment (dweindl) [reply pypesto/visualize/profiles.py:20]: "or something similar" :sweat_smile: 
I am not completely sure which types are supported.

**Pull request merged**